recipes/mold/all/conanfile.py

```diff
@@ -0,0 +1,75 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.45.0"
+
+class MoldConan(ConanFile):
+    name = "mold"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/rui314/mold/"
+    license = "Zlib"
+    description = ("mold is a faster drop-in replacement for existing Unix linkers. It is several times faster than the LLVM lld linker")
+    topics = ("mold", "ld", "linkage", "compilation")
+
+    settings = "os", "arch", "compiler", "build_type"
+
+    generators = "make"
+
+    def validate(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration('{} can\'t be built on {}.'.format(self.name, self.settings.os))
+        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) <= "10":
+            raise ConanInvalidConfiguration('Use Gcc compiler version higher than 10')
+        if (self.settings.compiler == "clang" or self.settings.compiler == "apple-clang") and tools.Version(self.settings.compiler.version) <= "11":
+            raise ConanInvalidConfiguration('Use Clang compiler version higher than 11')
+        if self.settings.compiler == "apple-clang" and "arm" in self.settings.arch :
+            raise ConanInvalidConfiguration('Use apple-clang does not work on ARM with this recipe')
```

Comment:
Is this correct? mold builds for arm on Debian just fine.

The error message could use better grammar as well.

Reply:
It fails on M1. I don't know, if it is a general mold problem or if it is conan ci problem.
I would have this PR merged and improve it in the follow-ups

Reply:
Please resolve